modal/_tunnel.py

```diff
@@ -61,10 +60,11 @@ def run_app():
     if not isinstance(port, int) or port < 1 or port > 65535:
         raise InvalidError(f"Invalid port number {port}")
 
-    if not config.get("task_id"):
-        raise InvalidError("Forwarding ports only works inside a Modal container")
+    if not client:
+        client = await _Client.from_env()
 
-    client = await _Client.from_env()
+    if client.client_type != api_pb2.CLIENT_TYPE_CONTAINER:
```

Comment:
might need to add a type annotation to the `_Client` class for `client_type`

Reply:
Thanks! I was scratching my head about that